pynbody/snapshot/tipsy.py

```diff
@@ -1181,61 +1169,115 @@ def __init__(self, filename, sort=True, paramfile=None):
-            molecH = False
+                                                   'f8', 'f8', 'f8','f8')})
+                molecH = True
+                # Unfortunately molecularH with small iOrders has the same as
+                # no moleculuarH with big iOrders.  Attempt to distinguish here
+                if(iSize == file_structure.itemsize):
+                    if(self._byteswap):
+                        testread = np.fromstring(
+                            f.read(iSize), dtype=file_structure).byteswap()
+                    else:
+                        testread = np.fromstring(f.read(iSize), dtype=file_structure)
+                    # All star iorders are greater than any gas iorder
+                    # so this indicates a bad format. (N.B. there is the
+                    # possibility of a false negative)
+                    if(testread['iord'][0] < testread['iorderGas'][0]): 
+                        file_structure = np.dtype({'names': ("iord", "iorderGas",
+                                                 "tform",
+                                                 "x", "y", "z",
+                                                 "vx", "vy", "vz",
+                                                 "massform", "rhoform", "tempform"),
+                                           'formats': ('i8', 'i8', 'f8',
+                                                       'f8', 'f8', 'f8',
+                                                       'f8', 'f8', 'f8',
+                                                       'f8', 'f8', 'f8')})
+                        f.seek(4)
+                        logger.info("Using 64 bit iOrders")
+                        molecH = False
+                        bigIOrds = True
+            if (iSize != file_structure.itemsize):
+                file_structure = np.dtype({'names': ("iord", "iorderGas", "tform",
+                                                     "x", "y", "z",
+                                                     "vx", "vy", "vz",
+                                                     "massform", "rhoform", "tempform",
+                                                     "phiform", "nsmooth"),
+                                           'formats': ('i4', 'i4', 'f8',
+                                                       'f8', 'f8', 'f8',
+                                                       'f8', 'f8', 'f8',
+                                                       'f8', 'f8', 'f8',
+                                                       'f8', 'i4')})
+                molecH = False
+
+                if (iSize != file_structure.itemsize and iSize != 104):
+                    raise IOError("Unknown starlog structure iSize:" + \
+                        str(iSize) + ", file_structure itemsize:" + \
+                        str(file_structure.itemsize))
+                else:
+                    bigstarlog = True
+            if molecH == True: print("h2 information found in StarLog!")
```

Comment:
This should probably be `logger.info` rather than `print`

Reply:
While I agree that this belongs in `logger.info` rather than as `print`, this print statement existed before, along with, e.g.,  [this one](https://github.com/pynbody/pynbody/blob/cfbefac64bfad3b8d042e669ff2b36bad25813ce/pynbody/snapshot/tipsy.py#L945). Do you want it addressed here, or in a separate dedicated PR to cleaning up `print`s that should be `logger.info`s?

Reply:
If you are willing to do a separate PR cleaning up these other instances, that would be much appreciated